frameworks/Julia/Http.jl/server.jl

```diff
@@ -1,7 +1,13 @@
+using Pkg
+
+Pkg.activate(@__DIR__)
+
 using HTTP
-using JSON
+import JSON3
 using Dates
+using Sockets
 
+@info "starting listener"
 HTTP.listen("0.0.0.0", 8080, reuseaddr=true) do http
```

Comment:
FWIW, I think this is all you need to make this multithreaded:

```jl
Threads.@threads for _ in 1:Threads.nthreads()
    HTTP.listen("0.0.0.0", 8080, reuseaddr=true) do http
        # blah
    end
end
```

Reply:
Thanks! I tried this both with the multiple processes and without and both were significantly slower than the current state - any theories as to why?

Reply:
Sorry for the bad steer! I'm just skimming the code in HTTP.jl, but nothing is screaming out to me.

If you're doing multiprocess with SO_REUSEPORT (which you are if `reuseaddr=true` and you're on a \*nix), then the kernel can handle all the shared state (working out which stream is for which listener) for you, and maybe whatever Julia is doing in threaded mode is getting in the way of that?

Or it could be something to do with your OS. Maybe the process ulimits aren't high enough, so 8 threads sharing the same limits are losing out when compared to 8 processes with independent ulimits. No real idea. Just guesses.